plat/kvm/arm/efi_arm.c

```diff
@@ -0,0 +1,42 @@
+#include <kvm/efi.h>
+#include <uk/arch/paging.h>
+#include <uk/plat/common/bootinfo.h>
+#include <uk/plat/lcpu.h>
+#include <uk/plat/common/sections.h>
+
+extern void clean_and_invalidate_dcache_range(unsigned long, unsigned long);
+extern void _ukplat_entry(void *) __noreturn;
+extern __paddr_t vector_table;
+extern void start_mmu(void);
+
+static __u8 __align(16) uk_efi_bootstack[__PAGE_SIZE];
+
+/* We cannot set the system register directly */
+#define uk_efi_set_sp_el1(sp)						\
+	__asm__ __volatile__("mov sp, %0\n" ::"r" (sp))
+
+void __noreturn uk_efi_jmp_to_kern(void)
```

Comment:
How is that different than the existing `entry64.S`? Should we be using unified entry code?

Reply:
Did you mean `lcpu_start.S`? `entry64.S` does not really make sense to me as it does some `virt` specific stuff. If you did mean `lcpu_start`, then yes it should work and I took into consideration. Basically that function does not touch some registers I touch here, so it would work if I touch those registers here and then jump to `lcpu_start` to let it touch the others. The reason why I decided to keep everything here is because I would only save 3 lines of code if I called `lcpu_start`. Furthermore, if I have everything here instead, it makes it much more clear what it takes to bridge the gap between UEFI and Unikraft, IMHO. In the future, as things get merged between architectures and code looks more uniform, I might even merge the two architectures' UEFI pre-kernels. For now, I would like to keep it this way. The way it is now I do not like it at all either... the main culprit for this though, I believe, is the lack of a unified, properly written IRQ subsystem.